platform/ios/src/MGLMapboxEvents.m

```diff
@@ -429,10 +424,15 @@ - (void)startUpdatingLocation {
     }
 }
 
+static BOOL turnstileEventSent = NO;
```

Comment:
Instead of this global static variable or a property on MGLMapboxEvents, `-pushTurnstileEvent` should be wrapped in a `dispatch_once()` block guarded by a static local variable. More broadly, we should eliminate all this complexity around thread safety with #3717.